cdap-app-fabric/src/main/java/org/apache/hadoop/mapred/LocalDistributedCacheManagerWithFix.java

```diff
@@ -107,7 +107,7 @@ public void setup(JobConf conf) throws IOException {
     MRApps.setupDistributedCache(conf, localResources);
     // Generating unique numbers for FSDownload.
     AtomicLong uniqueNumberGenerator =
-      new AtomicLong(System.currentTimeMillis());
+      new AtomicLong(System.nanoTime());
```

Comment:
The uniqueNumberGenerator is used for creating the unique job directories for the MapReduce programs. By adding fork, it is possible to trigger the multiple MapReduce programs at exactly same millisecond, which would cause both programs accessing the same directory. This is similar to the issue - https://issues.apache.org/jira/browse/HADOOP-10924
Added `System.nanoTime()` as a temporary measure. We can hit the same problem but only if the two programs are started at the same nanosecond which would be rare.